### ISSUE TITLE
How to map git remote to GitHub owner?

### ISSUE THREAD

**Issue by OliverJAsh** [github/hub#2235]
I am trying to write a command that generically opens a GitHub compare for a fork branch (related: https://github.com/github/hub/issues/1295).

So far I've got this:

``` sh
BRANCH=`git branch --show-current`
REMOTE=`git config branch.$BRANCH.remote`
hub compare $REMOTE:$BRANCH
```

However, the branch remote name doesn't correspond 1:1 with the GitHub fork owner/user (which is what we need for `hub compare`).

Does `hub` provide any means of mapping a git remote to a GitHub user? E.g. if my remote `me` has the URL `https://github.com/OliverJAsh/hub.git`, I would like to extract `OliverJAsh`.

**Comment by mislav** [comment]
No, I don't think hub provides a mechanism like this right now, but I have long felt that it should, mostly that it can be used in scripting like in your use-case.

**Comment by whuang8** [comment]
What do you folks think about extending the `remote` command to include
```sh
$ hub remote owner me
OliverJAsh
```
